src/lib.rs

```diff
@@ -3653,8 +3653,42 @@ impl Build {
     }
 
     fn apple_sdk_root(&self, sdk: &str) -> Result<OsString, Error> {
-        if let Some(sdkroot) = env::var_os("SDKROOT") {
-            return Ok(sdkroot);
+        // Code copied from rustc's compiler/rustc_codegen_ssa/src/back/link.rs.
+        if let Ok(sdkroot) = env::var("SDKROOT") {
```

Comment:
Let's use var_os here to handle non-utf8 path:

```suggestion
        if let Some(sdkroot) = env::var_os("SDKROOT") {
```

Reply:
We don't have `contains()` in OsString, which we need. An alternative that would avoid converting it to UTF-8 would be to parse it into a Path